src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -519,6 +519,8 @@ private static string GenerateUpgradeReport(string projectName, string backupPat
                         nuGetUI?.RecommendedCount,
                         nuGetUI?.RecommendPackages,
                         nuGetUI?.RecommenderVersion,
+                        nuGetUI?.TopLevelVulnerablePackagesCount ?? 0,
+                        nuGetUI?.TopLevelVulnerablePackagesMaxSeverity ?? -1,
```

Comment:
What does `-1` mean? It is not defined in https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#vulnerabilities

Reply:
It means, no nulverability information found

Reply:
It means: no vulnerable info found. If found this definition in DetailControlModel: 
https://github.com/NuGet/NuGet.Client/blob/4ba39b98697e715fd09ce762f59c27dc0809eea6/src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs#L416

Reply:
I don't like introducing literals all over the code. My preference would be to use an `enum` rather than an `int`. Is there a strong reason not to do so? Is the cost of doing that too much for this PR? 

If you agree it'd be good to change, but you can't now, maybe a follow-up issue to refactor the vulnerability implementation could be written?

Reply:
Changing _all_ NuGet vulnerabilities code would require to change NuGet.Protocol starting at:

https://github.com/NuGet/NuGet.Client/blob/c54c88a7a3eecb49f0e5b9f1c14fc73a8aba1f94/src/NuGet.Core/NuGet.Protocol/Model/PackageVulnerabilityMetadata.cs#L15

I can add a literal for No vulnerability (-1) and follow up on refactor to use an enum.

Reply:
Follow up issue: https://github.com/NuGet/Home/issues/11081